### PR TITLE
[SPARK-15261][SQL] Remove experimental tag from DataFrameReader/Writer

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch removes experimental tag from DataFrameReader and DataFrameWriter, and explicitly tags a few methods added for structured streaming as experimental.
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by rxin** [comment]
cc @davies

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -540,7 +538,9 @@ def partitionBy(self, *cols):
     def queryName(self, queryName):
         """Specifies the name of the :class:`ContinuousQuery` that can be started with
         :func:`startStream`. This name must be unique among all the currently active queries
-        in the associated SQLContext.
+        in the associated SQLContext
+
+        ::Note: Experimental.
```
Comment (davies) [inline python/pyspark/sql/readwriter.py]: .. note::

**Comment by rxin** [comment]
OK updated.

**Comment by davies** [comment]
LGTM

**Comment by rxin** [comment]
Merging in master/2.0.

**Pull request closed without merging**